other-guides/env-files.md

```diff
@@ -0,0 +1,151 @@
+
+If you don't add your `.env` to the `.gitignore` file then you risk your bot token being compromised. Don't let it come to that. The proper security measures.
+
+If you follow this properly, all users will see in your bot code is `process.env.ENV_VARIABLE` which exposes nothing!
+
+## Using Heroku
+Heroku is a website that allows you to start out and host your applications for free. In this case, your Discord bot. However, Heroku requires that you use environment variables. If you setup your files with the `dotenv` package and required the specific environment variables in your code, then all you have to do is go the environment variables in your Heroku application and add the key and value.
+
+In Heroku, these variables are called "Config Vars". I'm not going to go in-depth here about Heroku, but the same way you setup your `.env` file, you would add in new environment variables in Heroku.
+
+![Heroku Config Vars](https://i.imgur.com/MSmEO5K.png)
+
+The only way your bot token can be exposed along with other environment variables is if you do one or more of these things:
+- a) You accidentally expose your `.env` file because you didn't add it to `.gitigonre`
+- b) You added a collaborator through Heroku. A collaborator has almost full permissions as the app owner
+
+Both a and b can be controlled. It's just you need to be smart.
+
+## Set environment variables in the start script
+When starting your application, either locally on your computer via the command line, in a npm start script in your `package.json` or even in a `Dockerfile` you can set what environment your bot should run in. For example, you may want to run your bot in production on Heroku or Glitch and in development on your computer. You can simply do that via adding new scripts in your `package.json`.
+
+Upon doing `npm init` when you first made your bot, you should have seen a `test` script created. The scripts portion of your `package.json` should look like this if you added nothing.
+```json
+"scripts": {
+  "test": "echo \"Error: no test specified\" && exit 1",
+},
+```
+
+Here, you can add multiple scripts. Where going to add a few scripts. `production`, `development`, and `start`.
+```json
+"scripts": {
+  "test": "echo \"Error: no test specified\" && exit 1",
+  "start": "node .",
+  "production": "NODE_ENV=production&&npm start",
+  "development": "set NODE_ENV=development&&npm start"
+},
+```
+- To start your bot in a production environment, you would do `npm run production`. This will set `process.env.NODE_ENV` to `production`
+- To start your bot in a development environment, you would do `npm run development`. This will set `process.env.NODE_ENV` to `development`
+
+In your code, you can define what should happen depending on the environment loaded. Here's an example I use in relation to DiscordBots.org in my `app.js`:
```

Comment:
Please do not reference third party services such as bot listing websites, we are not affiliated with them, and I do not want to mislead, indirectly or directly users to thinking we're affiliated.

Reply:
Agreed, the dbots parts should not be in the guide. It should be sufficient to show how the env variables are used - how they're used beyond discord.js isn't relevant to this guide. Well. not much anyway.

Reply:
I will change this surely. I wasn't sure of the specific rules here, so I thought I mind as well shoot for it and wait for a response from one of you two (Will resolve the conversation once the update has been made).